packages/office-ui-fabric-react/src/components/Icon/Icon.base.tsx

```diff
@@ -1,103 +1,91 @@
 /* tslint:disable */
 import * as React from 'react';
 /* tslint:enable */
-import { IIconProps, IconType } from './Icon.types';
+import { IIconProps, IconType, IIconStyleProps, IIconStyles } from './Icon.types';
 import { Image } from '../Image/Image';
 import { ImageLoadState } from '../Image/Image.types';
 import {
   css,
   getNativeProps,
   htmlElementProperties,
-  BaseComponent
+  BaseComponent,
+  classNamesFunction
 } from '../../Utilities';
 import { getIcon } from '../../Styling';
-import { getClassNames } from './Icon.classNames';
 
 export interface IIconState {
   imageLoadError: boolean;
 }
 
-export class Icon extends BaseComponent<IIconProps, IIconState> {
+const getClassNames = classNamesFunction<IIconStyleProps, IIconStyles>();
+
+export class IconBase extends BaseComponent<IIconProps, IIconState> {
   constructor(props: IIconProps) {
     super(props);
     this.state = {
       imageLoadError: false,
     };
```

Comment:
Do you still need to add warnDeprecations to alert that iconName will be removed?

Reply:
I kind of want to do that in 6.0. Icons are used everywhere and warning a kajillion times is going to generate noise. My proposal is that we use `process.env==='production'` checks around warn apis to delete the code in prod builds, but that could break the odsp AMD scenarios...